lib/hanami/slice.rb

```diff
@@ -948,24 +948,44 @@ def load_router(inspector:)
 
         require_relative "slice/router"
 
+        slice = self
         config = self.config
         rack_monitor = self["rack.monitor"]
 
         Slice::Router.new(
           inspector: inspector,
           routes: routes,
           resolver: config.router.resolver.new(slice: self),
+          not_allowed: ROUTER_NOT_ALLOWED_HANDLER,
+          not_found: ROUTER_NOT_FOUND_HANDLER,
           **config.router.options
         ) do
           use(rack_monitor)
-          if Hanami.bundled?("hanami-controller")
-            use(*config.actions.sessions.middleware) if config.actions.sessions.enabled?
+
+          use(
+            Hanami::Middleware::RenderErrors,
+            config,
+            Hanami::Middleware::PublicErrorsApp.new(slice.root.join("public"))
+          )
```

Comment:
I decided to keep this middleware inserted at all times — even when `config.render_errors` is false — because it helpers to provide a "stable" default middleware stack that our users can depend upon for inserting their middleware at the right spot.

And since this is an exception-catching middleware, it's actually quite important for user-provided to middleware to know whether they're inserting themselves before or after this one.

In terms of how this works when `config.render_errors` is false, it simply re-raises the exception in that case.

Given that this middleware is likely to be enabled for the majority of apps in production mode, having it still exist (but do nothing) in development/test modes seems like a worthwhile tradeoff of the predictable middleware stack.